Mathlib/MeasureTheory/MeasurableSpace/CountablyGenerated.lean

```diff
@@ -99,6 +99,128 @@ instance [MeasurableSpace α] [CountablyGenerated α] [MeasurableSpace β] [Coun
     CountablyGenerated (α × β) :=
   .sup (.comap Prod.fst) (.comap Prod.snd)
 
+variable (α)
+
+theorem exists_seq_generateFrom [m : MeasurableSpace α] [h : CountablyGenerated α] :
+    ∃ e : ℕ → Set α, m = generateFrom (range e) := by
+  rcases countable_countableGeneratingSet.exists_eq_range $
+    nonempty_countableGeneratingSet (α := α) with ⟨e, he⟩
+  use e
+  rw [← he, generateFrom_countableGeneratingSet]
+
+variable {α}
+
+section SeparatesPoints
+
+/-- We say that a measurable space separates points if for any two distinct points,
+there is a measurable set containing one but not the other. -/
+class SeparatesPoints (α : Type*) [m : MeasurableSpace α] : Prop where
+  separates : ∀ x y : α, (∀ s, MeasurableSet s → (x ∈ s ↔ y ∈ s)) → x = y
```

Comment:
Can you state this definition in a way that mimics `Set.SeparatesPoints`? That is, with the contraposition of that implication. We have `Set.SeparatesPoints`, `Subalgebra.SeparatesPoints` and now `MeasurableSpace.SeparatesPoints` and they are all essentially the same statement, more or less specialized. It would be nice if they looked similar.

Also please add API for the definition. For example, for two points x, y not equal, we want a definition that gives us a set with x in it and not y, and a lemma stating that it is measurable.

If we change the definition of SeparatesPoints at I suggest, we will also want your current formulation of `separates` as a lemma.

Reply:
Hmm, I was intentionally mimicking `HasCountableSeparatingOn`, and I do like that this way of writing it involves less contraposing down the line. Do you still think I should change it?

Reply:
Yes, I understand and apologize for not adding more supporting lemmas initially. I'll get right on that.